app/addons/components/react-components.react.jsx

```diff
@@ -1081,6 +1105,114 @@ function (app, FauxtonAPI, React, Components, ace, beautifyHelper) {
 
   });
 
+  var ToggleState = React.createClass({
+    getDefaultProps: function () {
+      return {
+        defaultLeft: 'true'
+      };
+    },
+    render: function () {
+      var config = this.props.toggleConfig,
+          defaultLeft = false,
+          defaultRight = false;
+
+      var title = config.title,
+          leftLabel = config.leftLabel,
+          rightLabel = config.rightLabel,
+          leftClick = config.leftClick,
+          rightClick = config.rightClick,
+          enclosingID = config.enclosingID;
+
+      if (config.defaultLeft) {
+        defaultLeft = "checked";
+        defaultRight = false;
+      } else {
+        defaultRight = "checked";
+        defaultLeft = false;
+      }
+
+      return (
+        <div id={enclosingID} className="toggle-states">
+          <div className="toggle-title noselect">{title}</div>
+          <div className="toggles">
+            <input type="radio" 
+              id = {"toggle-state-left-id-" + enclosingID}
+              name={"toggle_" + enclosingID}
+              className="input-toggle-hidden"
+              defaultChecked = {defaultLeft}
+              onClick={leftClick} />
+              <label 
+                htmlFor={"toggle-state-left-id-" + enclosingID} 
+                className="checkbox-label toggle-state-button left noselect">
+                {leftLabel}
+              </label>
+            <input type="radio"
+              id = {"toggle-state-right-id-" + enclosingID} 
+              name={"toggle_" + enclosingID}
+              defaultChecked = {defaultRight}
+              className="input-toggle-hidden "
+              onClick={rightClick} />
+              <label 
+                htmlFor={"toggle-state-right-id-" + enclosingID}
+                className="checkbox-label toggle-state-button right noselect">
+                {rightLabel}
+              </label>
+            </div>
+        </div>
+      );
+    }
+  });
+
+  var SmallDropdown = React.createClass({
+    getInitialState: function () {
```

Comment:
Could we add a `propTypes` to this class? It's nice to know at a glance exactly what props are being passed into a class so there are no surprises.